### PR TITLE
[ent-search] Reorganize yamlRestTests into product-specific folders

### PR DESCRIPTION
**Pull request by jedrazb**
## Changes

Reorg `ent-search` yaml tests into `connector`, `search`, `analytics` and `rules` folders. Adapt the number prefix for each test.

Managing all the tests in the top-level folder was becoming unbearable.

### CODE REVIEW

File: x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_usage.yml
Comment (ioanatia) [inline x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_usage.yml:1]: why `20_usage.yml` and not `10_usage.yml`?

Comment (jedrazb) [reply x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_usage.yml:1]: We already have `10_basic.yml` - the only file I didn't touch
Screenshot 2024-04-22 at 13 55 33

**Review by ioanatia — COMMENTED** [review]
looks good on my end - I guess I would have grouped all the connectors in a single directory with subdirectories (e.g. `secret`, `sync_job`), but this approach is def an improvement from what we currently have.

**Comment by jedrazb** [comment]
> looks good on my end - I guess I would have grouped all the connectors in a single directory with subdirectories (e.g. secret, sync_job), but this approach is def an improvement from what we currently have.

@ioanatia It might be hard to see (given the number of files) but I did group all connector-related tests into `connector` dir with `sync_job` and `secret` as subdirs of `connector` dir. Or did you mean something else?

**Review by ioanatia — APPROVED** [review]

**Pull request merged**